### PR TITLE
Add resolve_hostname(..) util function, refactor some utils into sub-modules

### PR DESCRIPTION
**Pull request by whummer**
Add `resolve_hostname(..)` util function, refactor some utils into sub-modules.

Currently still leaving the imports in `common.py` to keep them accessible from other places - left a `TODO` to refactor the imports in the remaining codebase, so we can later remove the imports from `common.py`

### CODE REVIEW

**Review by dfangl — APPROVED** [review]
LGTM, just some comments, nothing critical though. Nice to see the splitup of the common module :tada:

File: tests/unit/utils/test_net_utils.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from localstack.constants import LOCALHOST
+from localstack.utils.common import short_uid
+from localstack.utils.net_utils import resolve_hostname
+
+
+def test_resolve_hostname():
+    assert "127.0." in resolve_hostname(LOCALHOST)
```
[the remark this answers is not in the document]
Comment (whummer) [reply tests/unit/utils/test_net_utils.py:7]: Great catch! Updated.. 👍

**Review by thrau — COMMENTED** [review]
the resolve_hostname function seems fine to me.

to the refactoring:

i guess it makes sense to organize utility methods into modules. however i'm not a fan of the current organization and naming into `localstack/utils/generic/<something>_utils.py`. the paths seem redundant and unnecessarily long. also it's not clear to me what makes them generic? what's less generic about the other utils?

wouldn't it be enough (and much cleaner) to just have something like:
* `localstack/utils/numbers.py`
* `localstack/utils/net.py`
* `localstack/utils/files.py`
* `localstack/utils/dicts.py`

etc..?

File: localstack/utils/net_utils.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        return False
+
+
+def is_ipv4_address(address: str) -> bool:
+    """
+    Checks if passed string looks like an IPv4 address
+    :param address: Possible IPv4 address
+    :return: True if string looks like IPv4 address, False otherwise
+    """
+    return bool(re.match(IP_REGEX, address))
```
[the remark this answers is not in the document]
Comment (whummer) [reply localstack/utils/net_utils.py:176]: Good point, totally agreed that less custom code / regexes on our end would be preferable. 👍  Interesting case for the class B / 32bit integer value - do you think you could pick this one up in one of the next iterations.. 🙌

**Comment by whummer** [comment]
> i guess it makes sense to organize utility methods into modules. however i'm not a fan of the current organization and naming into localstack/utils/generic/<something>_utils.py. the paths seem redundant and unnecessarily long. also it's not clear to me what makes them generic? what's less generic about the other utils?

Good point, totally agree @thrau - had the same thoughts when I started this refactoring, actually! 😄  So, question is whether we want to leave this to a follow-up iteration where we properly introduce a new structure like the one you've proposed (which could then also go hand-in-hand with a larger refactoring of the imports across the codebase). Agreed that this makes sense?

**Comment by thrau** [comment]
let's do a follow up PR, i'll create an issue for it :-)

File: localstack/utils/net_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dfangl) [reply localstack/utils/net_utils.py:176]: Will do :+1:

**Pull request merged**